### PR TITLE
update template to handle js and css arrays

### PR DESCRIPTION
**Pull request by digitalsadhu**
Adds support for js and css as arrays in template function in a backwards compatible way.
Adds a new footer template argument

### CODE REVIEW

File: lib/utils.js [hunk 1]
```diff
@@ -247,30 +247,45 @@ const deserializeContext = (headers = {}, prefix = 'podium') => {
...
+} = {}) => {
+    let scripts = js;
+    let styles = css;
+
+    const buildPreloadTag = ({ value = '' }) =>
+        `<link rel="preload" href="${value}" as="script"></link>`;
+    const buildScriptTag = ({ value = '' }) =>
+        `<script src="${value}" defer></script>`;
+    const buildCSSLinkTag = ({ value = '' }) =>
+        `<link media="all" rel="stylesheet" type="text/css" href="${value}">`;
```
Comment (stipsan) [inline lib/utils.js]: the `media` attribute has the value `all` by default, so I don't think we should add it.

For the same reason we don't add attributes with defaults, like a `type="text/javascript"` to the `<script>` tags 🙂

File: __tests__/__snapshots__/utils.js.snap [hunk 2]
```diff
@@ -1,19 +1,47 @@
...
+        <script src=\\"http://somejsurl2.com\\" defer></script>
+        <script src=\\"http://somejsurl3.com\\" defer></script>
+        
+    </body>
+</html>"
+`;
+
 exports[`.template() - arguments given - should render template using values given 1`] = `
 "<!doctype html>
 <html lang=\\"en-NZ\\">
```
Comment (trygve-lie) [inline __tests__/__snapshots__/utils.js.snap]: Ehem....

File: lib/utils.js [hunk 3]
```diff
@@ -247,30 +247,45 @@ const deserializeContext = (headers = {}, prefix = 'podium') => {
...
-                : ''
-        }
+        ${styles.map(buildCSSLinkTag).join('\n        ')}
+        ${scripts.map(buildPreloadTag).join('\n        ')}
         <title>${title}</title>
         ${head}
     </head>
     <body>
         ${body}
+        ${scripts.map(buildScriptTag).join('\n        ')}
```
Comment (trygve-lie) [inline lib/utils.js]: All scripts should go in the head.

File: __tests__/__snapshots__/utils.js.snap (hunk 2 above)
Comment (digitalsadhu) [reply __tests__/__snapshots__/utils.js.snap]: pretty sure it's a thing...

**Pull request merged**